### DISCUSSION TITLE
HERE Maps API Changes for New Keys

### DISCUSSION THREAD

**Discussion by mwegithub** [openlayers/openlayers#16512] [Show and tell]
I've used OpenLayers for many years to allow my users to access OpenStreetMap, CyclOSM, Bing Maps and HERE Maps.
Recently users reported that HERE Maps no longer work with HERE API keys they recently acquired. After some investigation and with the friendly help of a user named Mike on StackOverflow, I was able to solve the problem.

### In Short: Recent HERE API Keys Require V3 Endpoints.

Older API keys still work with the older (V2) and the new HERE V3 endpoints. But new keys require V3 endpoints.

I have described the changes I had to make to my code in the [StackOverflow discussion](https://stackoverflow.com/questions/79310028/openlayers-and-here-maps-with-recent-api-keyw-no-longer-working/79313546#79313546), in case somebody runs into the same problem.

The [official OpenLayers example](https://www.openlayers.vip/here-maps.html) uses the ```https://{1-4}.{base}.map...``` API url, which does not longer work with newer HERE API keys. The URL and parameters have changed for the V3 version of HERE Maps.
The changes required are in the StackOverflow post, maybe somebody can add a new example or update the existing example?

**Comment by ahocevar** [comment]
The HERE maps example has been removed from the official examples collection a while ago. The link you provided above is not part of the official [OpenLayers website](https://openlayers.org/).

**Comment by mwegithub** [comment]
I apologize, my bad.
